Library/Homebrew/cmd/update.sh

```diff
@@ -640,10 +631,31 @@ EOS
 
       # HOMEBREW_UPDATE_FORCE and HOMEBREW_UPDATE_AUTO aren't modified here so ignore subshell warning.
       # shellcheck disable=SC2030
-      if [[ "${UPSTREAM_REPOSITORY_URL}" == "https://github.com/"* ]]
+      if [[ "${UPSTREAM_REPOSITORY_URL}" = "https://github.com/"* ]] ||
+         [[ "${UPSTREAM_REPOSITORY_URL}" =~ https://(([^:@]+)(:([^@]+))?@)?github.com/(.*)$ ]]
       then
-        UPSTREAM_REPOSITORY="${UPSTREAM_REPOSITORY_URL#https://github.com/}"
-        UPSTREAM_REPOSITORY="${UPSTREAM_REPOSITORY%.git}"
+        if [[ -n "${BASH_REMATCH[5]}" ]]
```

Comment:
Again: this should be not be checked first if `[[ "${UPSTREAM_REPOSITORY_URL}" = "https://github.com/"* ]]`